### PR TITLE
Textarea input cleanup

### PR DESCRIPTION
**Pull request by protich**
Strip tags on text input field to prevent potential XSS exploit.

Addresses #2871

### CODE REVIEW

**Comment by protich** [comment]
@infectormp - Thanks for pointing it out, it was a back port mishap. Commit rebased to fix the issue.

File: include/class.thread.php [hunk 1]
```diff
@@ -1476,6 +1476,22 @@ static function fromFormattedText($text, $format=false) {
             return new ThreadBody($text);
         }
     }
+
+    static function clean($text, $format=null) {
```
Comment (greezybacon) [inline include/class.thread.php]: Shouldn't this replace `getClean`? The `ThreadEntry::create()` method uses `getClean` to determine if the entry is empty.

File: include/class.thread.php [hunk 2]
```diff
@@ -1484,7 +1500,8 @@ function __construct($body, $options=array()) {
     }
 
     function getClean() {
-        return Format::stripEmptyLines($this->body);
+        return  Format::stripEmptyLines(
+                self::clean($this->body, $this->format));
```
Comment (greezybacon) [inline include/class.thread.php]: Based on the above comment, seems like this should become
```php
    return Format::stripEmptyLines(parent::getClean($this->body));
```

Comment (greezybacon) [inline include/class.thread.php]: Isn't the format stored in `$this->type`?

File: include/class.thread.php (hunk 1 above)
Comment (greezybacon) [inline include/class.thread.php]: Perhaps
```php
function getClean() {
       switch ($this->type) {
       case 'html':
           return trim($this->body, " <>br/\t\n\r") ? $this->body: '';
       case 'text':
           return trim($this->body) ? $this->body: '';
       default:
           return $this->body;
       }
}

static function clean($text, $format=null) {
    $format = $format ?: ($cfg->isHtmlThreadEnabled() ? 'html' : 'text');
    $body = static::fromFormattedText($text, $format);
    return $body->getClean();
}
```

File: include/class.thread.php [hunk 3]
```diff
@@ -1527,7 +1544,7 @@ function ($m) use ($self) {
     }
 
     function getClean() {
-        return trim($this->body, " <>br/\t\n\r") ? Format::sanitize($this->body) : '';
+        return Format::sanitize(self::clean($this->body, $this->format));
```
Comment (greezybacon) [inline include/class.thread.php]:
```php
function getClean() {
    return Format::sanitize(parent::getClean());
}
```

**Pull request merged**